code/game/objects/items/card_trimmer.dm

```diff
@@ -0,0 +1,36 @@
+/obj/item/card_trimmer
+	name = "card trimmer"
+	desc = "A device that can replace the trim of Nanotrasen cards, changing its predefined access slots."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "card_trimmer"
+	w_class = WEIGHT_CLASS_SMALL
+
+	var/selected_trim = NONE
+	var/list/valid_trims = list(TRIM_SERVICE, TRIM_SECURITY, TRIM_MEDICAL, TRIM_SCIENCE, TRIM_ENGINEERING, TRIM_SUPPLY, TRIM_COMMAND)
+
+/obj/item/card_trimmer/attack_self(mob/user)
+	var/list/options = new/list()
+		options["None"] = NONE // get_region_accesses_name returns "all" for NONE, which doesn't make sense in this context
+	for(var/trim in valid_trims)
+		options["[get_region_accesses_name(trim)]"] = trim // So they show up as the region name, not the define number.
+	var/choice = input(user, "Select a Trim", "Card Trimmer", NONE) as null|anything in options
+	selected_trim = options[choice]
+	return
```

Comment:
```suggestion
	var/choice = input(user, "Select a Trim", "Card Trimmer", NONE) as null|anything in options
	if(isnull(choice))
		return
	selected_trim = options[choice]

```
* Empty `return` at the end of the proc is not really needed.
* If you give the cancel option (with `null|anything`) then it's expected you'll cancel the action if that's selected.
* Consider using a radial menu here.